### DISCUSSION TITLE
Studio Performance Issues with Version 4.2

### DISCUSSION THREAD

**Discussion by EsatMan** [stride3d/stride#2168] [Q&A] [answered]
Hello everyone,

I recently started using the Stride game engine, having installed version 4.2. After creating a project using the default template, I encountered a recurring issue where the software intermittently causes significant system lag. This lag is frequent enough to render the entire computer unresponsive, including other applications, and even the mouse interaction with the desktop becomes impossible.

Notably, the lag occurs even when I am not interacting with Stride (i.e., when the window is not in focus and I am using other applications). During these lag episodes, I observed a sudden spike in Stride's CPU usage, approximately 20%, in the Task Manager. Following the lag, the CPU usage returns to normal levels.

I did some search but find nothing about "lag" on the Internet, and I'm not sure what is causing the problem, so I came for help.

Here are the specifications of my Windows environment:

- Windows Edition: Windows 10 Professional
- Version: 22H2
- Installation Date: September 16, 2023
- OS Build: 19045.4046
- Feature Experience Pack: 1000.19053.1000.0
- 4K desktop

Hardware Specifications:

- CPU: 13th Gen Intel(R) Core(TM) i9-13900K @ 3.00 GHz
- RAM: 64.0 GB
- GPU: NVIDIA GeForce RTX 4090

If anyone has experienced similar problems or has insights into potential solutions, it would be greatly appreciated.

Thank you for your time and assistance.

![state1](https://github.com/stride3d/stride/assets/69354697/244f0e7a-cde0-4e01-a0ee-8fb83b35fa91)
![state2](https://github.com/stride3d/stride/assets/69354697/00447600-3a7c-41f5-8cc9-c42f544435f6)

**Comment by EsatMan** [comment]
Upgraded to Win11 and the issue seems to be irreproducible. While it's a kind of solution I suppose.

**Comment by FloccinaucinihilifilipicatesLongNames** [comment, marked as answer]
By lag, you mean random freezes for a few seconds? If so, I've been experiencing the same issue for a while but really didn't had this happen in my previous PC with a Ryzen5 CPU until I upgraded to a 12th gen i9, same Win10 installation. It only seems to happen when rendering anything on my end.

If it helps any, I normally bandaid-fix it by changing the CPU affinity of the Stride processes in Task Manager under _Details_ to only use the last few cores. Somewhat tedious to do it everytime I restart the app or the editor but gets the job done.

**Reply by EsatMan** [reply]
Thanks for reply, I do mean freezes, and the answer helps a lot.

**Reply by FloccinaucinihilifilipicatesLongNames** [reply]
You're welcome, good to know the answer was helpful to you. 👍🏻
